server/src/models/types.ts

```diff
@@ -0,0 +1,245 @@
+export type entity = userModel | addressModel | countryModel | stateModel | cityModel | cartModel | number | string;
+
+interface loginDetail {
+    userEmail: string,
+    userPassword: string
+}
+export class loginDetailModel {
+    user_email: string;
+    user_password: string;
+    constructor(input: loginDetail) {
+        this.user_email = input.userEmail;
+        this.user_password = input.userPassword;
+    }
+}
+
+interface user {
+    userId?: number,
+    userName: string,
+    userEmail: string,
+    userPassword: string,
+    userPhoneNumber: string
+}
+export class userModel {
+    user_id?: number;
+    user_name: string;
+    user_email: string;
+    user_password: string;
+    user_phone_number: string;
+    constructor(input: user) {
+        this.user_id = input.userId;
+        this.user_name = input.userName;
+        this.user_email = input.userEmail;
+        this.user_password = input.userPassword;
+        this.user_phone_number = input.userPhoneNumber;
+    }
+}
+
+interface address {
+    addressId?: number,
+    address: string,
+    city: string,
+    state: string,
+    zipCode: number,
+    country: string
+}
+export class addressModel {
+    address_id?: number;
+    address: string;
+    city: string;
+    state: string;
+    zip_code: number;
+    country: string;
+    constructor(input: address) {
+        this.address_id = input.addressId;
+        this.address = input.address;
+        this.city = input.city;
+        this.state = input.state;
+        this.zip_code = input.zipCode;
+        this.country = input.country;
+    }
+}
+
+interface userInfo extends user, address { };
+export class userInfoModel {
+    user_id?: number;
+    user_name: string;
+    user_email: string;
+    user_password: string;
+    user_phone_number: string;
+    address_id?: number;
+    address: string;
+    city: string;
+    state: string;
+    zip_code: number;
+    country: string;
+    constructor(input: userInfo) {
+        this.user_id = input.userId;
+        this.user_name = input.userName;
+        this.user_email = input.userEmail;
+        this.user_password = input.userPassword;
+        this.user_phone_number = input.userPhoneNumber;
+        this.address_id = input.addressId;
+        this.address = input.address;
+        this.city = input.city;
+        this.state = input.state;
+        this.zip_code = input.zipCode;
+        this.country = input.country;
+    }
+}
+
+type country = {
```

Comment:
Avoid making types with classes, types are generally used when an value can have more than one type.

Reference :

interface ICountryRequest {
    countryId: number,
    countryName: string
}

interface ICountry {
    country_id: number,
    country_name: string,
    getName() : string,
}

export class CountryModel implements ICountry {
    country_id: number;
    country_name: string;
    constructor(input: ICountryRequest) {
        this.country_id = input.countryId;
        this.country_name = input.countryName;
    }
    public getName(){
        return this.country_name;
    }
}